### PR TITLE
[Fix #420] Recommend YARD instead of RDoc for API documentation

### LINKED ISSUE #420: TomDoc or Yard instead of Rdoc
The style-guide says:

```
Use RDoc and its conventions for API documentation.
```

However, the lack of consistent structure and readability in RDoc have lead to the specification of TomDoc. The style-guide should perhaps encourage TomDoc (optional)?

Here is some relevant reading:

http://tom.preston-werner.com/2010/05/11/tomdoc-reasonable-ruby-documentation.html
http://gnuu.org/2010/05/12/whats-missing-from-tomdoc/
http://tomdoc.org/

### PR DESCRIPTION
**Pull request by conradbeach**
The general consensus in issue #420 was that it would be preferable to recommend YARD instead of RDoc for API documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -695,10 +695,10 @@ Translations of the guide are available in the following languages:
   num = 1234
   ```
 
-* <a name="rdoc-conventions"></a>
-    Use [Rdoc][rdoc] and its conventions for API documentation.  Don't put an
+* <a name="api-documentation"></a>
+    Use [YARD][yard] and its conventions for API documentation.  Don't put an
```
Comment (bbatsov) [inline README.md]: Probably we can remove the second sentence here, as it seems unrelated to me.

**Comment by bbatsov** [comment]
I'm fine with the change, although I wonder whether docs should be in the scope of the guide. Alternatively we can just remove this entry completely (I think this was suggested a few times as well).

File: README.md (hunk 1 above)
Comment (conradbeach) [reply README.md]: Ok. I'll remove the second sentence.
